src/lib/localStorage.spec.ts

```diff
@@ -0,0 +1,47 @@
+import { getLocalStorageItem, setLocalStorageItem, removeLocalStorageItem } from "./localStorage";
+
+describe("localStorage tests", () => {
+  beforeEach(() => {
+    localStorage.clear();
+    jest.clearAllMocks();
+  });
+
+  test("getLocalStorageItem not existing", () => {
+    expect(getLocalStorageItem("test")).toBeUndefined();
+  });
+
+  test("getLocalStorageItem existing", () => {
+    localStorage.setItem("test", '{"data":{"field1":"hello","field2":"world"}}');
+    expect(getLocalStorageItem("test")).toMatchObject({ field1: "hello", field2: "world" });
+  });
+
+  test("getLocalStorageItem expired", () => {
+    const expiredDate = new Date(new Date().getTime() - 1);
+    localStorage.setItem("test", `{"data":true,"expirationDate":"${expiredDate.toISOString()}"}`);
+    expect(localStorage.length).toBe(1);
+    expect(getLocalStorageItem("test")).toBeUndefined();
+    expect(localStorage.length).toBe(0);
+  });
+
+  test("setLocalStorageItem without expiration", () => {
+    setLocalStorageItem("test", { field1: "hello", field2: "world" });
+    expect(localStorage.length).toBe(1);
+    expect(localStorage.getItem("test")).toBe('{"data":{"field1":"hello","field2":"world"}}');
+  });
+
+  test("setLocalStorageItem with expiration", () => {

```

Comment:
suggestion:

```
test("setLocalStorageItem stores data with expiration", () => {
    const expirationDate = new Date(2050, 1, 1, 16, 30, 45, 123);
    setLocalStorageItem("test", true, expirationDate);

    // Remove the system timezone to properly verify the serialized time
    const expectedExpirationDate = new Date(expirationDate.getTime() - expirationDate.getTimezoneOffset() * 60000);

    expect(localStorage.length).toBe(1);
    expect(localStorage.getItem("test")).toBe(JSON.stringify({ data: true, expirationDate: expectedExpirationDate.toISOString() }));
  });
```

Reply:
I've had to remove the "hack" that removes the timezone since using JSON.stringify also for the expected value it applies the timezone so there's no need to remove it